community/kernel/src/main/java/org/neo4j/kernel/impl/api/TransactionRepresentationCommitProcess.java

```diff
@@ -62,59 +62,65 @@ public long commit( TransactionRepresentation transaction, LockGroup locks, Comm
     {
         try ( ValidatedIndexUpdates indexUpdates = validateIndexUpdates( transaction ) )
         {
-            // serialise commands to the log
-            long transactionId = commitTransaction( transaction, commitEvent );
-            commitEvent.setTransactionId( transactionId );
-
-            // apply changes to the store
-            try ( StoreApplyEvent storeApplyEvent = commitEvent.beginStoreApply() )
-            {
-                storeApplier.apply( transaction, indexUpdates, locks, transactionId, mode );
-            }
-            // TODO catch different types of exceptions here, some which are OK
-            catch ( Throwable e )
-            {
-                throw exception( CouldNotCommit, e,
-                        "Could not apply the transaction to the store after written to log" );
-            }
-            finally
-            {
-                transactionIdStore.transactionClosed( transactionId );
-            }
-
+            long transactionId = serialiseToLog( transaction, commitEvent );
+            applyToStore( transaction, locks, commitEvent, indexUpdates, transactionId );
             return transactionId;
         }
     }
 
-    private TransactionFailureException exception( Status status, Throwable cause, String message )
+    private ValidatedIndexUpdates validateIndexUpdates(
+            TransactionRepresentation transaction ) throws TransactionFailureException
     {
-        kernelHealth.panic( cause );
-        return new TransactionFailureException( status, cause, message );
+        try
+        {
+            return indexUpdatesValidator.validate( transaction, mode );
+        }
+        catch ( Throwable e )
+        {
+            throw new TransactionFailureException( ValidationFailed, e, "Validation of index updates failed" );
+        }
     }
 
-    private long commitTransaction( TransactionRepresentation tx, CommitEvent commitEvent ) throws TransactionFailureException
+    private long serialiseToLog(
```

Comment:
Might be unnecessary to introduce an additional term `serialise` when we have `append`.